webapp/channels/src/components/root/performance_reporter_controller.tsx

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useRef} from 'react';
+import {useStore} from 'react-redux';
+
+import {Client4} from 'mattermost-redux/client';
+
+import PerformanceReporter from 'utils/performance_telemetry/reporter';
+
+export default function PerformanceReporterController() {
+    const store = useStore();
+
+    const reporter = useRef<PerformanceReporter>();
+
+    useEffect(() => {
+        reporter.current = new PerformanceReporter(Client4, store);
```

Comment:
whats the reasoning behind using ref here? If store wasnt the dependency would we have used constant instead?

Reply:
It's using a ref because we only want one instance of this, and I didn't want this to be a global singleton like we do with `store` and `Client4`. I've been trying to avoid singletons lately since I feel like people misuse them sometimes,